cartographer/io/fake_file_writer.cc

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/io/fake_file_writer.h"
+
+#include "glog/logging.h"
+
+namespace cartographer {
+namespace io {
+
+FakeFileWriter::FakeFileWriter(const std::string& filename,
+                               std::shared_ptr<std::string> content)
+    : is_closed_(false), content_(content), filename_(filename) {
+  CHECK(content);
```

Comment:
Consistency in the Cartographer code base: Explicitly check for null pointers for readability, i.e. `content != nullptr`.

Reply:
Done